src/get-changelog-pull-requests.js

```diff
@@ -0,0 +1,113 @@
+    const previousLog = await getCommits(octokit, {repo, owner, base: sharedBase.sha, head: previous});
+    const previousReleaseBranchDate = sharedBase.commit.committer.date;
+    const pullRequestsSince = await fetchMergedPullRequestsSinceDate(octokit, {
+        repo, owner, since: previousReleaseBranchDate
+    });
+    const pullRequestsInBranch = [];
+
+    for (const commit of currentLog) {
+        const includedPr = pullRequestsSince.find((pr) => {
+            return pr.merge_commit_sha === commit.sha;
+        });
+
+        const previouslyCherryPicked = detectCherryPicksOfCommitWithinLog(commit, previousLog);
+
+        if (includedPr && !previouslyCherryPicked) {
+            pullRequestsInBranch.push(includedPr);
+        }
+    }
+
+    const categorized = categorizePullRequests(pullRequestsInBranch); 
+    return categorized;
+};
+
+// Equivalent to `git log base..head`
+async function getCommits(octokit, {owner, repo, base, head}) {
+    let response = await octokit.repos.compareCommits({
+        owner,
+        repo,
+        base,
+        head,
+        headers: { accept: 'application/vnd.github.v3.sha' }
+    });
+
+    let {commits} = response.data;
+    if (response.data.total_commits > commits.length) {
+        console.error(`Warning: ${head} is ahead of ${base} by ${response.data.total_commits}, but only ${commits.length} are included in the result.`);
+    }
+
+    return commits;
+}
+
+async function findMergeBase(octokit, {owner, repo, base, head}) {
+    const response = await octokit.repos.compareCommits({
+        owner,
+        repo,
+        base,
+        head,
+        headers: { accept: 'application/vnd.github.v3.sha' }
+    });
+
+    return response.data.merge_base_commit;
+}
+
+async function fetchMergedPullRequestsSinceDate(octokit, {repo, owner, since}) {
+    let response = await octokit.pullRequests.getAll({
+        owner,
+        repo,
+        state: 'closed',
+        sort: 'updated',
+        direction: 'desc',
+        per_page: 100
+    });
+    let {data} = response;
+    while (data[data.length - 1].updated_at > since && octokit.hasNextPage(response)) {
+        response = await octokit.getNextPage(response);
+        data = data.concat(response.data);
+    }
+
+    const pullRequests = [];
+    for (const pr of data) {
+        if (pr.merge_commit_sha && pr.merged_at && pr.merged_at >= since) {
+            pullRequests.push(pr);
+        };
+    }
+
+    return pullRequests;
+}
+
+// Heuristic used to detect cherry picks.
+// Search for a commit message as a substring within a log of commits.
+function detectCherryPicksOfCommitWithinLog(commit, log) {
+    const found = log.some((logCommit) => {
+        return logCommit.commit.message.includes(commit.commit.message);
+    });
+
+    return found;
+}
+
+function categorizePullRequests(pullRequests) {
+    const hasChangelog = []
+    const skipChangelog = [];
+    const needsChangelog = [];
+
+    for (const pr of pullRequests) {
```

Comment:
The way I read the labels is to indicate whether the PR should or should not be included in the CHANGELOG. Where the label doesn't exist, it should default to inclusion in the draft (maybe with an 🤷‍♀ , to indicate lack of label)

A PR with the `needs changelog` label may or may not have a useful PR title or changelog suggestion in the body.